lighthouse-core/runner.js

```diff
@@ -25,6 +25,32 @@ const fs = require('fs');
 const path = require('path');
 const url = require('url');
 
+function runAudit(audit, artifacts) {
+  const status = `Evaluating: ${audit.meta.description}`;
+
+  return Promise.resolve().then(_ => {
+    log.log('status', status);
+
+    // Return an early error if an artifact required for the audit is missing.
+    for (const artifactName of audit.meta.requiredArtifacts) {
+      // TODO: need pass-specific check of networkRecords and traces
```

Comment:
are there any examples here that used a trace from anything but `Audit.DEFAULT_PASS`? seems like we could just do it now and avoid the wasted work if answer is no